tests/utils.py

```diff
@@ -270,6 +275,57 @@ def get_app_or_url(test_item):
     return app_or_url
 
 
+def get_test_webhook_app():
+    """
+    Instantiate a local test application used for the prehook for user creation and deletion.
+    """
+    def webhook_request(request):
+        # Simulates a webhook url call
+        user = request.POST["user_name"]
+        temp_url = request.POST["temp_url"]
+
+        # Status is incremented to count the number of successful test webhooks
+        settings["webhook_status"] += 1
+        return Response("Successful webhook url with user " + user + " and temp_url " + temp_url)
+
+    def get_status(request):
+        # Returns the status number
+        return Response(str(settings["webhook_status"]))
+
+    def reset_status(request):
+        settings["webhook_status"] = 0
+        return Response("Webhook status has been reset to 0.")
+
+    with Configurator() as config:
+        settings = config.registry.settings
+        # Initialize status
+        settings["webhook_status"] = 0
+        config.add_route("webhook", "/webhook")
+        config.add_route("get_status", "/get_status")
+        config.add_route("reset_status", "/reset_status")
+        config.add_view(webhook_request, route_name="webhook", request_method="POST", request_param="user_name")
+        config.add_view(get_status, route_name="get_status", request_method="GET")
+        config.add_view(reset_status, route_name="reset_status", request_method="POST")
+        webhook_app_instance = config.make_wsgi_app()
+
+    def webhook_app():
+        try:
+            webhook_url_info = urlparse(get_constant("MAGPIE_TEST_USER_WEBHOOK_URL"))
+            serve(webhook_app_instance, host=webhook_url_info.hostname, port=webhook_url_info.port)
+        except OSError as exception:
+            if exception.errno == EADDRINUSE:
+                # The app is already running, we just need to reset the webhook status for a new test.
+                resp = requests.post(get_constant("MAGPIE_TEST_USER_WEBHOOK_URL") + "/reset_status")
+                check_response_basic_info(resp, 200, expected_method="POST")
+                return
+            raise
+
+    x = threading.Thread(target=webhook_app, daemon=True)
+    x.start()
+
+    return webhook_app_instance
+
```

Comment:
All the `webhook_app` and handling of serve/threading can be simplified with `TestApp` using same process as in `get_test_magpie_app`. 
Doesn't need extra `waitress` and all other util functions will be available directly.

Reply:
@fmigneault 
The problem I had for creating the webhook app is that the context was a bit different. 
For the Magpie app in the tests, we create a TestApp, and we then access it directly to execute the multiple requests required.
For the webhook app, it has to be accessible via an URL from inside the Magpie app, so I couldn't use the TestApp approach, since that app would not be accessible where the webhooks requests would be called. I had to make it available via an URL, using waitress.